src/Build/Instance/ProjectItemInstance.cs

```diff
@@ -1576,9 +1579,8 @@ public bool Equals(TaskItem other)
                     return true;
                 }
 
-                // Since both sides are this class, we know both sides support ITaskItem2.
-                ITaskItem2 thisAsITaskItem2 = this as ITaskItem2;
-                ITaskItem2 otherAsITaskItem2 = other as ITaskItem2;
+                ITaskItem2 thisAsITaskItem2 = this;
```

Comment:
Why bother making these at all?

Reply:
The class implements `ITaskItem2` explicitly and explicit interface members can only be accessed via the interface type, even if you're "in" the implementing class.